src/app/model/migrations/migrations/to0002/MigrationUtils0002.ts

```diff
@@ -0,0 +1,78 @@
+import { Dirs, FileSystem } from 'react-native-file-access';
+
+import {
+  ActivityItemType,
+  ActivityState,
+  MediaFile,
+  PipelineItem,
+} from './MigrationStorageTypes0002';
+import { Storages } from '../../types';
+import {
+  getStorageRecord,
+  getStorageRecordKeys,
+  upsertStorageRecord,
+} from '../../utils';
+
+export const getActivityState = (key: string): ActivityState | null =>
+  getStorageRecord(Storages.ActivityProgress, key);
+
+export const updateActivityState = (key: string, state: ActivityState) => {
+  upsertStorageRecord(Storages.ActivityProgress, key, state);
+};
+
+export const getActivityStorageKeys = () =>
+  getStorageRecordKeys(Storages.ActivityProgress);
+
+const getFilenameFromLocalUri = (localUri: string) => {
+  const regex = /^file:\/\/(?:.*\/)?([^/]+)$/;
+  const match = localUri.match(regex);
+
+  return match ? match[1] : '';
+};
+
+export const evaluateLocalFileUri = (fileName: string) =>
+  `file://${Dirs.CacheDir}/${fileName}`;
+
+export const moveMediaFileToCache = async (
+  fileName: string,
+  fromUri: string,
+) => {
+  const localFileUri = evaluateLocalFileUri(fileName!);
+
+  let fileExists = await FileSystem.exists(localFileUri);
+
+  if (fileExists) {
```

Comment:
How is it possible that file exist in the re-named, current cache directory during migration 2 run?
Should migration be runned in this case?
Should we delete files in such cases?
Shoudn't we make check if file exist for fromUrl at the start of this function?

Reply:
What is migration 2 run?

Reply:
The function was copy-pasted from the mediaFiles.ts for safety reasons as we would not want to make different implementations of the same function in migration scripts.
Answering your question, it is not possible that the file already exists in cacheFileUri during the migration.